static/app/views/explore/tables/tracesTable/fieldRenderers.tsx

```diff
@@ -57,72 +57,58 @@ export function SpanDescriptionRenderer({span}: {span: SpanResult<Field>}) {
 interface ProjectsRendererProps {
   projectSlugs: string[];
   maxVisibleProjects?: number;
+  visibleAvatarSize?: number;
 }
 
 export function ProjectsRenderer({
   projectSlugs,
+  visibleAvatarSize,
   maxVisibleProjects = 2,
 }: ProjectsRendererProps) {
   const organization = useOrganization();
+  const {projects} = useProjects({slugs: projectSlugs, orgId: organization.slug});
+  const projectAvatars =
+    projects.length > 0 ? projects : projectSlugs.map(slug => ({slug}));
+  const numProjects = projectAvatars.length;
+  const numVisibleProjects =
+    maxVisibleProjects - numProjects >= 0 ? numProjects : maxVisibleProjects - 1;
+  const visibleProjectAvatars = projectAvatars.slice(0, numVisibleProjects).reverse();
+  const collapsedProjectAvatars = projectAvatars.slice(numVisibleProjects);
+  const numCollapsedProjects = collapsedProjectAvatars.length;
```

Comment:
@Abdkhan14 mind making this a function outside of the component and adding some comment behind the intentions here?

Reply:
@JonasBa I'll address this in a seperate PR. This is existing explore code, I removed the out dated Projects provider and used the `useProjects` hook instead, thus moving this block of code out to the component level.